File: Medium/Python/921. Minimum Add to Make Parentheses Valid.py
class Solution(object):
    def minAddToMakeValid(self, s):
        start = 0
        end = 0

        for i in range(len(s)):
            if s[i] == '(':
                end += 1
            else:
                if end != 0:
                    end -= 1
                else:
                    start += 1
        
        return start + end

File: Medium/JavaScript/921. Minimum Add to Make Parentheses Valid.js
var minAddToMakeValid = function(s) {
    let start = 0;
    let end = 0;

    for(let i = 0; i < s.length; i++){
        if(s.charAt(i) == '('){
            end++;
        }else{
            if(end != 0){
                end--;
            }else{
                start++;
            }
        }
    }

    return start + end;
};
